File: Semestru 3/MAP/lab/lab2/src/factory/TaskContainerFactory.java
package factory;

import container.Container;
import container.StackContainer;
import container.QueueContainer;

public class TaskContainerFactory implements Factory {
    private static TaskContainerFactory instance = new TaskContainerFactory();

    private TaskContainerFactory() {}

    public static TaskContainerFactory getInstance() {
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        switch (strategy) {
            case Strategy.FIFO: {
                return new QueueContainer();
            }
            case Strategy.LIFO: {
                return new StackContainer();
            }
            default: throw new IllegalArgumentException("Strategy not recognised " + strategy);
        }
    }
}

File: Semestru 3/MAP/lab/lab9/lab9/lab9/factory/TaskContainerFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using lab9.container;

namespace lab9.factory
{
	class TaskContainerFactory : IFactory
	{
		private static TaskContainerFactory instance = new TaskContainerFactory();
		private TaskContainerFactory() { }

		public static TaskContainerFactory getInstance() => instance!;

		public IContainerr createContainer(Strategy strategy)
		{
			switch (strategy)
			{
				case Strategy.FIFO:
					{
						return new QueueContainer();
					}
				case Strategy.LIFO:
					{
						return new StackContainer();
					}
				default: throw new Exception("Strategy not recognised " + strategy);
			}
		}
	}
}
